syntax = "proto3";

package pixanalyzer;

message AiAnalyzeCpfRequest {
  string id = 1;
  int32 origin_client_id = 2;
  string destination_key_value = 3;
  double amount = 4;
  string timestamp = 5;
  string key = 6;
  string key_type = 7;
  string institution = 8;
  string branch = 9;
  string account_number = 10;
  string account_type = 11;
  string owner_type = 12;
  string owner_name = 13;
  string cpf = 14;
  string full_name = 15;
  string birth_date = 16;
  string registration_date = 17;
  string status = 18;
  int32 common_transfers_client = 19;
  int32 all_transfers = 20;
}

message AiAnalyzeCpfResponse {
  double confidence_score = 1;
  repeated string fraud_reasons = 2;
}

service AiAnalyzeCpfService {
  rpc AiAnalyzeCpf (AiAnalyzeCpfRequest) returns (AiAnalyzeCpfResponse);
}